dfusion_rust_core/src/database/mod.rs

```diff
@@ -7,6 +7,9 @@ use super::models;
 pub use error::*;
 pub use graph_reader::GraphReader;
 
+use mockall::automock;
+
+#[automock]
```

Comment:
```suggestion
#[cfg_attr(test, automock)]
```

Reply:
Unfortunately this causes compilation errors, as we are using the MockDbInterface in unit tests of a different crate (the driver crate).

I wasn't able to tell the driver crate that when it's compiled in test mode, to also compile its dependencies in test mode. Thus I had to make this class available outside of dev mode.

Do you have a workaround for that?

Reply:
A solution like in the answer here https://stackoverflow.com/questions/41700543/can-we-share-test-utilites-between-crates *Use a testing-feature across all crates* might work. But it doesn't feel good to have to add an extra feature for this and always specify it when running tests. That answer is 3 years old so maybe there is a better way now.
I don't like having to add a feature for this and having to remember to use it when testing.

Alternatively https://docs.rs/mockall/0.6.0/mockall/#external-traits could work as it allows creating mocks for things not in the same crate.
I don't like this solution because it would lead to a lot of boiler plate.

Based on this the best way imo is to have the `automock` unconditional and not just as a dev-dependency in the cases where this is needed because simplicity is more important.

Reply:
I think a testing feature is probably the way to go here.

Reply:
Here is a gist on how this could be done without much hassle. Note that gists don't allow directories, so just `s/_/\//`:
https://gist.github.com/nlordell/12a6c46ff26d30f034b061b052bc9c2f

Reply:
Guess there is no good way around this :(